pkg/bootstrapper/bootstrapper.go

```diff
@@ -271,50 +278,135 @@ func (wmcb *winNodeBootstrapper) translateFile(ignitionSource string, fileTransl
 	return newContents, err
 }
 
+// extractEnvVarsFromSystemdUnit extracts a map of the environment variables present in a systemd unitfile
+func extractEnvVarsFromSystemdUnit(contents string) map[string]string {
+	envVars := make(map[string]string)
+	lines := strings.Split(contents, "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, EnvironmentPrefix) {
+			// The format of each environment variable line will look like this:
+			// Environment=\"KUBELET_LOG_LEVEL=3\"
+			quoteSplit := strings.Split(line, "\"")
+			if len(quoteSplit) != 3 {
+				// continue on error
+				continue
+			}
+			keyValSlice := strings.Split(quoteSplit[1], "=")
+			if len(keyValSlice) != 2 {
+				continue
+			}
+			envVars[keyValSlice[0]] = keyValSlice[1]
+		}
+	}
+	return envVars
+}
+
+// getEnvVarName returns the name of an environment variable located in the given string. The variable must be specified
+// by the format ${var_name}. There can only be one environment variable in the given string.
+func getEnvVarName(sequence string) (string, error) {
+	envVarSplit := strings.Split(sequence, "${")
+	if len(envVarSplit) != 2 {
+		return "", fmt.Errorf("could not find exactly one opening sequence")
+	}
+	trimSplit := strings.Split(envVarSplit[1], "}")
+	if len(trimSplit) != 2 {
+		return "", fmt.Errorf("could not find exactly one closing brace")
+	}
+	return trimSplit[0], nil
+}
+
+// resolveKubeletArg cleans the input value returns the true value of it, either unchanged or resolved via the provided
+// map
+func resolveKubeletArg(value string, envVars map[string]string) (string, error) {
+	// Line can contain an escaped newline which leaves the substring with a trailing "\"
+	partiallyTrimmed := strings.TrimSuffix(value, "\\")
+	trimmedValue := strings.TrimSpace(partiallyTrimmed)
+	if varName, err := getEnvVarName(trimmedValue); err == nil {
+		// arg value is an environment variable so use the environment variable value
+		varValue, ok := envVars[varName]
+		if !ok {
+			return "", fmt.Errorf("arg is set to a env variable that was not set")
+		}
+		trimmedValue = varValue
+	}
+	return trimmedValue, nil
+}
+
+// parseSystemdKubeletUnit gathers the kubelet args we need from the kubelet systemd unit file and returns a
+// map of file translations to be added to the existing file translation map
```

Comment:
I don’t understand the doc. Not only are you updating the file translation map but also other kubelet args. Please clarify. This fiction can be split into one that is handling file and non file arguments.